pandas/core/frame.py

```diff
@@ -8670,7 +8670,10 @@ def _reduce(
             cols = self.columns[~dtype_is_dt]
             self = self[cols]
 
-        any_object = self.dtypes.apply(is_object_dtype).any()
+        any_object = np.array(
+            [is_object_dtype(values.dtype) for values in self._iter_column_arrays()],
+            dtype=bool,
+        ).any()
```

Comment:
let's only find the dtypes once (i.e. share with dtype_is_dt above

```
own_dtypes = [arr.dtype for arr in self._iter_column_arrays()]
# or 
own_dtypes = [blk.dtype for blk in self._mgr.blocks]
```

Reply:
done in 41827fb